packages/store/src/state-factory.ts

```diff
@@ -101,28 +101,29 @@ export class StateFactory {
   invokeActions(getState, setState, dispatch, actions$, action) {
     const results = [];
 
-    for (const metadata of this.states) {
+    for (const data of this.states) {
+      const metadata: MetaDataModel = data;
+
       const name = getActionTypeFromInstance(action);
-      const actionMetas = metadata.actions[name];
+
+      const actionMetas = metadata.actions[name.toString()];
 
       if (actionMetas) {
         for (const actionMeta of actionMetas) {
           const stateContext = this.createStateContext(getState, setState, dispatch, metadata);
           let result = metadata.instance[actionMeta.fn](stateContext, action);
 
-          if (result === undefined) {
+          if (!result) {
             result = of({}).pipe(shareReplay());
           } else if (result instanceof Promise) {
             result = fromPromise(result);
           }
 
-          if (result instanceof Observable) {
-            result = result.pipe(
-              (<ActionOptions>actionMeta.options).takeLast
-                ? takeUntil(actions$.pipe(ofAction(action.constructor)))
-                : map(r => r)
-            ); // act like a noop
-          }
+          result = result.pipe(
+            (<ActionOptions>actionMeta.options).takeLast
+              ? takeUntil(actions$.pipe(ofAction(action.constructor)))
```

Comment:
Potential issue: if action is not a class, which happens when dev tools emit actions.

Reply:
it would just be undefined then. I don't think it will break